components/org.wso2.carbon.identity.authenticator.saml2.sso/src/main/java/org/wso2/carbon/identity/authenticator/saml2/sso/util/Util.java

```diff
@@ -156,11 +157,13 @@ public static X509CredentialImpl getX509CredentialImplForTenant(String domainNam
         }
 
         KeyStoreManager keyStoreManager = null;
-        // get an instance of the corresponding Key Store Manager instance
-        keyStoreManager = KeyStoreManager.getInstance(tenantID);
-
         X509CredentialImpl credentialImpl = null;
         try {
+            IdentityTenantUtil.initializeRegistry(tenantID, domainName);
```

Comment:
Do we need to do this for each request?

Reply:
Do we need to do this for super tenant?

Reply:
> Do we need to do this for each request?
If the tenant is loaded it will not loaded again. So this is ok.
https://github.com/wso2/carbon-identity-framework/blob/master/components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityTenantUtil.java#L215

Reply:
> Do we need to do this for super tenant?

Super tenanat will be escaped
https://github.com/wso2/carbon-identity-framework/blob/master/components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityTenantUtil.java#L217